src/components/RecipesList.js

```diff
@@ -33,7 +36,7 @@ class RecipesList extends React.Component {
         let recipes = data.map((recipe) => {
           return {id:recipe.id, title:recipe.title, image:recipe.image};
         });
-        this.setState({recipes: recipes})
+        this.setState({recipes: recipes, showDataLoading: false})
```

Comment:
I agree with Ben. If it is not too much trouble separating it, would be better having a separate function for it.

Reply:
Cool.
I agree and we have discussed offline with rest of team.
Conclusion is that we will go ahead with this merge and be knowingly introduce some technical debt as a deadline is approaching and this refactor can happen in due course.